### ISSUE TITLE
Create state machine to continue training if spot instance gets interrupted

### ISSUE THREAD

**Issue by tylerwooten** [aws-deepracer-community/deepracer-on-the-spot#27]

**Comment by markrossatos** [comment]
It might be easier to do this with an Autoscaling Group.  e.g.
 - ASG lifecycle action on termination to update the existing config files (e.g. run.env would need an update): -
    - DR_LOCAL_S3_PRETRAINED=True
    - DR_LOCAL_S3_PRETRAINED_PREFIX=location of training
    - DR_LOCAL_S3_PRETRAINED_CHECKPOINT=last
 -  Copy them up to S3 (might want to avoid custom_files as potential conflict with people running multiple models)
 - ASG launch template / configuration would need to pick up updated config on any future launch versus first launch

**Comment by markrossatos** [comment]
@tylerwooten I've got this working now without needing AWS Step Functions, it's all handled by the ASG.  With some initial testing proving it's working over training of an hour or two injecting spot interruptions using Fault Injection Simulator.  I've branched off my graph and tables branch (as I was concerned about merge conflicts if I didn't) so won't submit the PR until you've dealt with that one (https://github.com/aws-deepracer-community/deepracer-on-the-spot/pull/52) and I've done some more extensive testing of training over multiple days.  Sneak peek available on my fork for now that testing is based on - https://github.com/markrossatos/deepracer-on-the-spot/tree/spot-asg
